src/manifolds/StiefelSubmersionMetric.jl

```diff
@@ -0,0 +1,293 @@
+@doc raw"""
+    StiefelSubmersionMetric{T<:Real} <: RiemannianMetric
+
+The submersion (or normal) metric family on the [`Stiefel`](@ref) manifold.
+
+The family, with a single real parameter ``α>-1``, has two special cases:
+- ``α = -\frac{1}{2}``: [`EuclideanMetric`](@ref)
+- ``α = 0``: [`CanonicalMetric`](@ref)
+
+The family was described in [^HüperMarkinaLeite2021]. This implementation follows the
+description in [^ZimmermanHüper2022].
+
+[^HüperMarkinaLeite2021]:
+    > Hüper, M., Markina, A., Leite, R. T. (2021)
+    > "A Lagrangian approach to extremal curves on Stiefel manifolds"
+    > Journal of Geometric Mechanics, 13(1): 55-72.
+    > doi: [10.3934/jgm.2020031](http://dx.doi.org/10.3934/jgm.2020031)
+[^ZimmermanHüper2022]:
+    > Ralf Zimmerman and Knut Hüper. (2022).
+    > "Computing the Riemannian logarithm on the Stiefel manifold: metrics, methods and performance."
+    > arXiv: [2103.12046](https://arxiv.org/abs/2103.12046)
+
+# Constructor
+
+    StiefelSubmersionMetric(α)
+
+Construct the submersion metric on the Stiefel manifold with the parameter ``α``.
+"""
+struct StiefelSubmersionMetric{T<:Real} <: RiemannianMetric
+    α::T
+end
+
+@doc raw"""
+    q = exp(M::MetricManifold{ℝ, Stiefel{n,k,ℝ}, <:StiefelSubmersionMetric}, p, X)
+    exp!(M::MetricManifold{ℝ, Stiefel{n,k,ℝ}, q, <:StiefelSubmersionMetric}, p, X)
+
+Compute the exponential map on the [`Stiefel(n,k)`](@ref) manifold with respect to the [`StiefelSubmersionMetric`](@ref).
+
+The exponential map is given by
+````math
+\exp_p X = \operatorname{Exp}\bigl(
+    -\frac{2α+1}{α+1} p p^\mathrm{T} X p^\mathrm{T} +
+    X p^\mathrm{T} - p X^\mathrm{T}
+\bigr) p \operatorname{Exp}\bigl(\frac{\alpha}{\alpha+1} p^\mathrm{T} X\bigr)
+````
+"""
+exp(::MetricManifold{ℝ,Stiefel{n,k,ℝ},<:StiefelSubmersionMetric}, ::Any...) where {n,k}
+
+function exp!(
+    M::MetricManifold{ℝ,Stiefel{n,k,ℝ},<:StiefelSubmersionMetric},
+    q,
+    p,
+    X,
+) where {n,k}
+    α = metric(M).α
+    T = Base.promote_eltype(q, p, X)
+    # TODO:
+    # - dispatch to exp! for α = -1/2 and α = 0 for efficiency
```

Comment:
I'm not convinced we should do this. Being able to compare is useful for testing.

Reply:
Yes, but they can I believe both be implemented more efficiently than this general implementation.

Reply:
yes, that is why they already are.